types/lodash/lodash-tests.ts

```diff
@@ -3401,10 +3401,22 @@ fp.now(); // $ExpectType number
 // _.flowRight
 {
     const fn1 = (n: number): number => 0;
+    const fn1Optional = (n?: number): number => 0;
     const fn2 = (m: number, n: number): number => 0;
     const fn3 = (a: number): string => "";
     const fn4 = (a: string): boolean => true;
```

Comment:
How can I add a test for the generic function issue (described in this PR's description)? Generics and pipe only work in 3.4.1, but these tests are ran against older versions.

``` ts
declare const identity: <T>(t: T) => T;

// Expected and actual type: `<T>(a1: T) => T`
const fn1 = pipe(
    identity,
    identity,
);

// Expected type: `<T>(a1: T) => T`
// Actual type (prior to this PR): `(a1: {}) => {}`
const fn2 = pipe(
    identity,
    value => identity(value),
);
```